src/tool.cpp

```diff
@@ -219,6 +219,17 @@ DigParams getDigParams(const ItemGroupList &groups,
 		}
 	}
 
+	//infostream<<"result_diggable="<<result_diggable<<std::endl;
+	//infostream<<"result_time="<<result_time<<std::endl;
+	//infostream<<"result_wear="<<result_wear<<std::endl;
+
+	if (time_from_last_punch < tp->full_punch_interval) {
```

Comment:
Travis points out that `time_from_last_punch` hasn't been declared anywhere in `getDigParams`. I'm not sure what is to be done here...

Reply:
looking at the [original commit](https://github.com/minetest/minetest/pull/6804/commits/a86b2755945075fe655ccf0e126eec308df3ac3d) it seems that `time_from_last_punch` did previously exist, apparently without error... maybe i messed up the rebase somehow?

Reply:
Aha, `time_from_last_punch` was removed (probably because it was unused?) two years ago by HybridDog in 345e104, see [blame](https://github.com/minetest/minetest/blame/master/src/tool.cpp#L174).

Reply:
Also, see [this](
https://github.com/minetest/minetest/commit/345e1041a204ca9d40a25b71ae239ce3275f0755#diff-b2e9183274c47c49c500e499eacd627eR219) - L265 should be changed from

```cpp
u16 wear_i = 65535.0f * result_wear;
```

to

```cpp
u16 wear_i = U16_MAX * result_wear;
```

Reply:
the most obvious course of action would appear to be adding the original `time_from_last_punch` code back in - shall i see if i can reconstruct it?